Use always id as identification on drag&drop form creator for relations

signedav opened the pull request:
## Description
Regarding this issue [18290](https://issues.qgis.org/issues/18290) when something else created a form with drag and drop than the attributesform in the vectorlayerproperties there could have been the danger, that attributesform destroys data. This because it always expected the relation-id as identification, but the name of the field was name. 

This name is removed in the constructor of `QgsAttributeEditorRelation` now and will be **filled automatically with the relation.id** - so there will be no problems with this anymore.

Additionally with this PR the handling is improved to have always a **"DisplayName" for labeling the items in the Drag&Drop** - usually these are the field or relation names. But they don't have to be unique anymore, means in case an external plugin creates them not uniquely it will still not break the project.

Fix #18290

signedav committed:
fix relation id confusion

if inserting a relation the id is used as name because it's used as identificator

signedav committed:
displayName in DnDTreeItem

a displayName defined on creation. Usually on fields the fieldname and on relations the relationname.
no other logical use for that. The DnDTreeItem.name is used as id and should be unique, not like displayName.

signedav committed:
styling stuff

signedav committed:
generated sip file

nyalldawson commented on src/core/qgsattributeeditorelement.h:
```diff
@@ -316,15 +316,35 @@ class CORE_EXPORT QgsAttributeEditorRelation : public QgsAttributeEditorElement
+    Q_DECL_DEPRECATED QgsAttributeEditorRelation( const QString &name, const QString &relationId, QgsAttributeEditorElement *parent )
+      : QgsAttributeEditorElement( AeTypeRelation, name, parent )
+      , mRelationId( relationId )
+      , mShowLinkButton( true )
```
Can these booleans be initialised at declaration to avoid the duplicate initialisation code in these constructors please?

signedav committed:
initialize variables on declaration

instead of initializing them on every constructor.
it's mShowLinkButton and mShowUnlinkButton

signedav commented:
Do you think it's okay now @nyalldawson ?
Thanks a lot.

nyalldawson commented:
Code style looks good to me - I'll let @m-kuhn review the change itself :)

m-kuhn commented:
The fix looks fine, thanks @nyalldawson and @signedav

signedav commented:
Thanks guys!
